Android/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.example.immersivereadersdk
+
+import android.app.Activity
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.webkit.CookieManager
+import android.webkit.WebView
+import android.widget.Button
+import android.webkit.WebViewClient
+import android.widget.LinearLayout
+import android.widget.TextView
+import com.google.gson.*
+import io.github.cdimascio.dotenv.dotenv
+import java.io.IOException
+import java.io.*
+import java.net.HttpURLConnection
+import java.net.HttpURLConnection.HTTP_OK
+import java.net.URL
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import java.util.*
+import kotlin.collections.ArrayList
+
+// This sample app uses the Dotenv is a module that loads environment variables from a .env file to better manage secrets.
+// https://github.com/cdimascio/java-dotenv
+// Be sure to add a "env" file to the /assets folder
+// instead of '.env', use 'env'
+
+class MainActivity : AppCompatActivity() {
+
+    private val dotEnv = dotenv {
+        directory = "/assets"
+        filename = "env"
+        ignoreIfMalformed = true
+        ignoreIfMissing = true
+    }
+
+    private lateinit var contextualWebView: WebView
+    private lateinit var messageData: Message
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        this.supportActionBar!!.hide()
+        setContentView(R.layout.activity_main)
+        val immersiveReaderButton = findViewById<Button>(R.id.LaunchImmersiveReaderButton)
+        immersiveReaderButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView() } }
+    }
+
+    private suspend fun handleLoadImmersiveReaderWebView() {
+        val exampleActivity = this
+        val subdomain = dotEnv["SUBDOMAIN"]
+        val irTitle = findViewById<TextView>(R.id.Title)
+        val irText1 = findViewById<TextView>(R.id.Content1)
+        val irText2 = findViewById<TextView>(R.id.Content2)
+        val appUtilities = AppUtilities(exampleActivity)
+
+        // The content of the request that's shown in the Immersive Reader.
+        // This basic example contains chunks of two different languages.
+        val chunk1 = Chunk(irText1.text.toString(), "en", "text/plain")
+        val chunk2 = Chunk(irText2.text.toString(), "fr", "text/plain")
+
+        val chunks = ArrayList<Chunk>()
+        chunks.add(chunk1)
+        chunks.add(chunk2)
+
+        val data = Content(irTitle.text.toString(), chunks)
+
+        // Options passed to the Immersive Reader

```

Comment:
i suggest only assigning the onExit callback in this sample since it's a quickstart, and comment a link to options documentation https://docs.microsoft.com/en-us/azure/cognitive-services/immersive-reader/reference#options

Reply:
Sounds good to me.